Add MCP conformance test server

trepel opened the pull request:
### Summary

Integrate the official ModelContextProtocol conformance test server to validate MCP protocol implementation in the gateway.

  Changes:
  - Add Dockerfile building from upstream conformance repo
  - Deploy as test-conformance-server with internal/external routes
  - Add workflow for image building
  - Configure with no toolPrefix (conformance test suite requirement)
  - Document in CLAUDE.md and test servers README

  The server uses the reference TypeScript implementation from
  github.com/modelcontextprotocol/conformance to ensure gateway
  compatibility with official MCP specifications.

### Notes
External route is created for the sake of consistency, it is not actually used.
I am not quite sure about `inspect-conformance-server` make target, added for the sake of consistency but none from the `inspect-*` make targets worked for me (`make: Nothing to be done for 'inspect-server1'.`)


### Testing & Verification

`make local-env-setup`
`make inspect-gateway`

Initialize, list the tools, call some conformance server tool (e.g. test_simple_text)

You can also call the conformance server test scenarios (https://github.com/modelcontextprotocol/conformance/tree/main?tab=readme-ov-file#testing-servers):
`npx @modelcontextprotocol/conformance server --url http://mcp.127-0-0-1.sslip.io:8001/mcp`

Results I got:
```
=== SUMMARY ===
✓ server-initialize: 1 passed, 0 failed
✗ logging-set-level: 0 passed, 1 failed
✗ completion-complete: 0 passed, 1 failed
✓ tools-list: 1 passed, 0 failed
✓ tools-call-simple-text: 1 passed, 0 failed
✓ tools-call-image: 1 passed, 0 failed
✓ tools-call-audio: 1 passed, 0 failed
✓ tools-call-embedded-resource: 1 passed, 0 failed
✓ tools-call-mixed-content: 1 passed, 0 failed
✗ tools-call-with-logging: 0 passed, 1 failed
✓ tools-call-error: 1 passed, 0 failed
✓ tools-call-with-progress: 1 passed, 0 failed
✗ tools-call-sampling: 0 passed, 1 failed
✗ tools-call-elicitation: 0 passed, 1 failed
✗ elicitation-sep1034-defaults: 0 passed, 1 failed
✗ resources-list: 0 passed, 1 failed
✗ resources-read-text: 0 passed, 1 failed
✗ resources-read-binary: 0 passed, 1 failed
✗ resources-templates-read: 0 passed, 1 failed
✗ resources-subscribe: 0 passed, 1 failed
✗ resources-unsubscribe: 0 passed, 1 failed
✗ prompts-list: 0 passed, 1 failed
✗ prompts-get-simple: 0 passed, 1 failed
✗ prompts-get-with-args: 0 passed, 1 failed
✗ prompts-get-embedded-resource: 0 passed, 1 failed
✗ prompts-get-with-image: 0 passed, 1 failed
```

Prompts and resources are not supported so the failures are expected. Not quite sure about the other failures (sampling, elicitation, logging, completion-complete).

david-martin commented on config/samples/mcpserver-test-servers.yaml:
```diff
@@ -115,6 +115,21 @@ spec:
+  labels:
+    'kagenti/mcp': 'true'
+spec:
+  # toolPrefix: conformance_ # conformance test suite expects no prefix
```
no prefix makes sense.
Noting I had issues with any tools that started with `test_` from the conformance server as they conflict with the test1 server which has a prefix of `test_`. 
Let's change that server to use test1_ prefix

david-martin commented:
Notes from a call with @trepel today:

* The failing test for the 'broken-server' looks unrelated to these changes. OK to defer for now (see other notes)
* Let's put this PR on hold as the functionality for supporting an MCPServer with no toolPrefix is buggy, and not covered by any e2e tests.
    * The [logic for stripping the server prefix](https://github.com/kagenti/mcp-gateway/blob/8665eaaa9ba666a961e0d9424c8549d1356ecf30/internal/config/mcpservers.go#L36-L49) is flawed. If a server has no prefix, any tool will match with that server. So tool calls intended for a specific server may end up being routed to a server with no tool prefix. The ordering of the servers causes non deterministic behaviour here as well. If the non prefixed server is first in the for loop, it will always match.
    * That logic is duplicated when [looking up the serverInfo](https://github.com/kagenti/mcp-gateway/blob/8665eaaa9ba666a961e0d9424c8549d1356ecf30/internal/config/mcpservers.go#L53-L68) [earlier in the flow](https://github.com/kagenti/mcp-gateway/blob/8665eaaa9ba666a961e0d9424c8549d1356ecf30/internal/mcp-router/request_handlers.go#L192-L226) (lookup server based on tool prefix). The serverInfo should be looked up once and referenced when needed.
* @trepel is going to PR a design doc that outlines how the non prefixed MCPServer feature should work. Let's review and agree on that design, implement it, then come back to this PR and rebase.

trepel commented:
@david-martin I have just added 3rd commit. Changes:
- introduced a separate make target for conformance server deployment
- removal of external route to conformance mcp server. I did not want to pollute gateway.yaml with external route listener if the conformance server is not deployed with other servers.

Please re-review if you can spare some time. Thanks!

trepel commented on Makefile:
```diff
@@ -198,6 +206,21 @@ deploy-test-servers: kind-load-test-servers ## Deploy test MCP servers for local
+	@sleep 3
+	@echo "Restarting broker to ensure connection..."
+	kubectl rollout restart deployment/mcp-broker-router -n mcp-system
+	@kubectl rollout status deployment/mcp-broker-router -n mcp-system --timeout=60s
```
Implemented in 4th commit.
